src/odemis/odemisd/modelgen.py

```diff
@@ -392,6 +475,11 @@ def _parse_instantiation_model(self, inst_file):
         """
         try:
             # The standard PyYAML loader is dangerous as it can create any python object
+
+            # Allow to include external yaml files via the keyword '!include' by adding a constructor.
+            SafeLoader.add_constructor('!include', SafeLoader.include)
+            # Overwrite construct_yaml_map with custom mapper
+            SafeLoader.yaml_constructors['tag:yaml.org,2002:map'] = SafeLoader.construct_yaml_map
```

Comment:
I think it's possible to move this just after the SafeLoader class. This way, it's done just once (on importing the module), and the caller code doesn't have to "baby-sit" the class.

Reply:
So you would implement it more or less like:

```
class SafeLoader(yaml.SafeLoader):
  def __init__(self, stream):
  .....
  def other_methods()
  ......
  def last_methods_class()
  ......

# Add keys to the Safeloader class
# Allow to include external yaml files via the key '!include' by adding a constructor.
SafeLoader.add_constructor('!include', SafeLoader.include)
# Overwrite construct_yaml_map with custom mapper which adds support for the "!extend" key
SafeLoader.yaml_constructors['tag:yaml.org,2002:map'] = SafeLoader.construct_yaml_map
```

Reply:
Yes, that's how it's also done in the YAML code.